eth/backend.go

```diff
@@ -87,12 +90,23 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 		method = "unknown"
 	}
 
+	txParams := make(map[string]interface{})
+	err = decodeTxParams(b.abiMap[string(data[:4])], txParams, data)
 	if err != nil {
```

Comment:
since we're only reading `err` within this `if` block, we can optimize as follows:
```go
if err = decodeTxParams(b.abiMap[string(data[:4])], txParams, data); err != nil {
    return err
}
```

Reply:
Good point. 

Is there a difference from a runtime perspective? As in is declaring a new err var more expensive than reassigning the existing err var or not?

Reply:
From a run-time perspective, it's pretty much the same.  It's a code readability improvement, mostly.  This scopes the return value of `decodeTxParams` to exist solely within the affirmative if case code block. The expense of allocating this error variable is negligible to non-existent - the compiler knows to optimize it.

Reply:
That's pretty clear, thanks ! 
fixed in d3504b70